### ISSUE TITLE
Query Runs Perfectly But Syntax Error Coloring

### ISSUE THREAD

**Issue by ufuk-ergin-carbon** [dbeaver/dbeaver#34538]
### Description

I am attaching a screenshot where the query runs perfectly. However, DBeaver highlights some parts in red. Any ideas why? 
![Screenshot 2024-06-27 193349](https://github.com/dbeaver/dbeaver/assets/114302087/6cb65acc-e10f-41bb-9971-2e09337622ed)

### DBeaver Version

Version 24.1.1.202406231636

### Operating System

Windows 11 Enterprise

### Database and driver

Server: Snowflake 8.20.0
Driver: Snowflake 3.14.4

**Comment by E1izabeth** [comment]
Thank you for bug report.

**Comment by E1izabeth** [comment]
I reproduced the issue. 

```
WITH results AS (
    SELECT res.id, res.HL7MESSAGEID
    FROM ANALYTICS_PROD_DB.ANALYTICS_BASE.MONOLITH_LABRESULTS res
    WHERE res.LABREQUESTID IS NOT NULL AND res.HL7MESSAGEID IS NOT NULL
        AND res.LABTESTNAME NOT IN ('General Comments', 'PDF Report')
        AND res.STATUS IN ('CorrectedResults', 'FinalResults', 'PreliminaryResults')  
)
SELECT
    results.hl7messageid,
    SUBSTR(
        me.message,
        POSITION('OBR|2' IN me.message) + LENGTH('OBR|2|'),
        POSITION('^' IN SUBSTR(me.message, POSITION('OBR|2' IN me.message) + LENGTH('OBR|2|'))) - 1
    ) AS extracted_part
FROM
    FIVETRAN_DB.POSTGRES_MONOLITH_PRODUCTION_PUBLIC.HL7MESSAGES me
INNER JOIN results ON results.HL7MESSAGEID = me.id
--LEFT JOIN ANALYTICS_PROD_DB.ANALYTICS_BASE.MONOLITH_TREATMENTPLANS tr
WHERE me.message LIKE '%OBR|2%';
```

**Comment by ufuk-ergin-carbon** [comment]
Hello @E1izabeth and @arhayka , I am not sure if what I experience after the last update (Version 24.1.4.202408041450) is related to this issue, but there is a query that runs perfectly, however I get "Column not found in dataset error" (next to exclamation mark). I am attaching the screenshots. Let me know if you need anything else.
![Screenshot 2024-08-06 140938](https://github.com/user-attachments/assets/22831845-b35a-471f-93a5-0e6374ae594c)
![Screenshot 2024-08-06 140217](https://github.com/user-attachments/assets/469b2c31-a2f5-4fa7-b051-06d490dee196)

**Comment by E1izabeth** [comment]
We added tooltips and error markers in DBeaver 24.1.4. Previously you saw only red highlighting in these cases, now you can read the reason.
I will look at it and let you know, if some additional information is needed.

**Comment by ufuk-ergin-carbon** [comment]
> We added tooltips and error markers in DBeaver 24.1.4. Previously you saw only red highlighting in these cases, now you can read the reason. I will look at it and let you know, if some additional information is needed.

OK, thank you for the update. Here in the issue, the query runs perfectly and therefore DBeaver has had to find the column in dataset in order to successfully execute the query. Not sure why it is breaking in syntax coloring.

**Comment by E1izabeth** [comment]
@ufuk-ergin-carbon unfortunately, I can't reproduce the behavior described in the comment https://github.com/dbeaver/dbeaver/issues/34538#issuecomment-2271024577
Is it stable reproducible for you? Could you try following actions to check if it helps:
- disconnect and then connect to the database
- put whitespace to any place in the the query to trigger reparsing
- change active catalog/schema

**Comment by ufuk-ergin-carbon** [comment]
> @ufuk-ergin-carbon unfortunately, I can't reproduce the behavior described in the comment [#34538 (comment)](https://github.com/dbeaver/dbeaver/issues/34538#issuecomment-2271024577) Is it stable reproducible for you? Could you try following actions to check if it helps:
> 
> * disconnect and then connect to the database
> * put whitespace to any place in the the query to trigger reparsing
> * change active catalog/schema

Hello @E1izabeth, I am confused. I thought you managed to reproduce the issue as per your comment above. Are you talking about reproducing the issue in [this comment](https://github.com/dbeaver/dbeaver/issues/34538#issuecomment-2271024577)?

**Comment by E1izabeth** [comment]
Yes. What you describe [here](https://github.com/dbeaver/dbeaver/issues/34538#issuecomment-2271024577) in the comment is a different issue and I can't reproduce it.

**Comment by ufuk-ergin-carbon** [comment]
OK, I'll try to reproduce the issue again.

**Comment by ufuk-ergin-carbon** [comment]
@E1izabeth, I tried all the steps you mentioned, and the issue is still there. Here is another screenshot where the query runs perfectly, but there is error syntax coloring all around. Let me know how I can assist more.

![Screenshot 2024-08-25 140617](https://github.com/user-attachments/assets/d8addf3f-ff2f-41a6-be18-928d2a546020)

**Comment by E1izabeth** [comment]
Thank you

**Comment by E1izabeth** [comment]
As a temporarily workaround, you can tun off `Read database table/columns for semantic analysis` in `Preferences->Editors->SQL Editor->Code Editor`

**Comment by E1izabeth** [comment]
Will be fixed in DBeaver 25.1.3
